cmake/modules/FindGMock.cmake

```diff
@@ -113,172 +117,329 @@ function(GTEST_ADD_TESTS executable extra_args)
-        set(GMOCK_INCLUDE_DIR  "${source_dir}/include")
+        set(GMOCK_INCLUDE_DIR "${source_dir}/include")
         mark_as_advanced(GMOCK_INCLUDE_DIR)
-        set(GMOCK_LIBRARY      "${binary_dir}/${CMAKE_CFG_INTDIR}/${CMAKE_FIND_LIBRARY_PREFIXES}gmock${Suffix}")
+        set(GMOCK_LIBRARY "${binary_dir}/${CMAKE_CFG_INTDIR}/${GTEST_LIBRARY_PREFIX}gmock${Suffix}")
         mark_as_advanced(GMOCK_LIBRARY)
-        set(GMOCK_MAIN_LIBRARY "${binary_dir}/${CMAKE_CFG_INTDIR}/${CMAKE_FIND_LIBRARY_PREFIXES}gmock_main${Suffix}")
+        set(GMOCK_MAIN_LIBRARY "${binary_dir}/${CMAKE_CFG_INTDIR}/${GTEST_LIBRARY_PREFIX}gmock_main${Suffix}")
         mark_as_advanced(GMOCK_MAIN_LIBRARY)
-    else() #1.8.0
-        ExternalProject_Add(
-            gmock
-            GIT_REPOSITORY "https://github.com/google/googletest.git"
-            GIT_TAG "release-${GMOCK_VER}"
-            PREFIX ${GMOCK_ROOT}
-            INSTALL_COMMAND ""
-            LOG_DOWNLOAD ON
-            LOG_CONFIGURE ON
-            LOG_BUILD ON
-            CMAKE_ARGS ${GTEST_CMAKE_ARGS}
-        )
+    endif()
+endif()
 
-        add_library(libgtest IMPORTED STATIC GLOBAL)
-        add_dependencies(libgtest gmock)
-        add_library(libgmock IMPORTED STATIC GLOBAL)
-        add_dependencies(libgmock gmock)
-        
-        ExternalProject_Get_Property(gmock source_dir binary_dir)
-        set(GTEST_INCLUDE_DIR  "${source_dir}/googletest/include")
-        set(GTEST_LIBRARY      "${binary_dir}/googlemock/gtest/${CMAKE_CFG_INTDIR}/${CMAKE_FIND_LIBRARY_PREFIXES}gtest${Suffix}")
-        set(GTEST_MAIN_LIBRARY "${binary_dir}/googlemock/gtest/${CMAKE_CFG_INTDIR}/${CMAKE_FIND_LIBRARY_PREFIXES}gtest_main${Suffix}")
-        set(GMOCK_INCLUDE_DIR  "${source_dir}/googlemock/include")
-        set(GMOCK_LIBRARY      "${binary_dir}/googlemock/${CMAKE_CFG_INTDIR}/${CMAKE_FIND_LIBRARY_PREFIXES}gmock${Suffix}")
-        set(GMOCK_MAIN_LIBRARY "${binary_dir}/googlemock/${CMAKE_CFG_INTDIR}/${CMAKE_FIND_LIBRARY_PREFIXES}gmock_main${Suffix}")
-        mark_as_advanced(GMOCK_LIBRARY)
-        mark_as_advanced(GMOCK_INCLUDE_DIR)
-        mark_as_advanced(GTEST_MAIN_LIBRARY)
-        mark_as_advanced(GTEST_LIBRARY)
-        mark_as_advanced(GTEST_INCLUDE_DIR)
-        mark_as_advanced(GMOCK_MAIN_LIBRARY)
-    endif() 
-else()
-   Message(STATUS "Building Gmock / Gtest from dir ${GMOCK_SRC_DIR}")
-   ExternalProject_Add(
-        gmock
-        SOURCE_DIR ${GMOCK_SRC_DIR}
-        PREFIX ${GMOCK_ROOT}
-        INSTALL_COMMAND ""
-        LOG_DOWNLOAD OFF
-        LOG_CONFIGURE ON
-        LOG_BUILD ON
-        CMAKE_ARGS ${GTEST_CMAKE_ARGS}
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(GTest DEFAULT_MSG GTEST_LIBRARY GTEST_INCLUDE_DIR GTEST_MAIN_LIBRARY)
+find_package_handle_standard_args(GMock DEFAULT_MSG GMOCK_LIBRARY GMOCK_INCLUDE_DIR GMOCK_MAIN_LIBRARY)
+
+include(CMakeFindDependencyMacro)
+find_dependency(Threads)
```

Comment:
Yes as far as I know you can not use find_package in FindXXX modules. This FindGMock.cmake is an adapted version of a recent FindGTest.cmake file plus the code from @konserw and this how its done in the FindGTest.cmake.